### DISCUSSION TITLE
Adding field value as data property

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#4815] [Help] [answered]
In my app I am saving money values as integers in the database (cent values) to prevent rounding errors. On screen those values are converted to decimals. But I would like to have the raw integer values attached to fields as well to do calculations. This _would_ work with extraInputAttributes:

`TextInput::make('price')->extraInputAttributes(['data-raw-value' => ???])`

How could I put a field's value into a data property?

**Comment by danharrin** [comment, marked as answer]
`->extraInputAttributes(fn ($state): array => ['data-raw-value' => $state])`

**Reply by danharrin** [reply]
https://filamentphp.com/docs/2.x/forms/advanced#using-closure-customization

**Reply by OvalMedia** [reply]
Thx, still have not gotten used to php8s language constructs.
There is a caveat though:
```
TextInput::make('price')
	->extraInputAttributes(fn ($state): array => ['data-raw-value' => $state])
	->afterStateHydrated(fn ($component, $state): TextInput => $component->state(round($state / 100, 2)))
])
```
`afterStateHydrated` already modified the `$state` value. Is there another function that modifies the value after extraInputAttributes?

**Reply by danharrin** [reply]
Maybe just use an input mask? https://filamentphp.com/docs/2.x/forms/fields#input-masking

Beware, these dont work inside repeaters or modals. But everywhere else.

**Reply by OvalMedia** [reply]
Unfortunately I am inside a repeater. If it does not exist I would suggest a feature to always be able to access the raw and unmodified original data from the database.
